### PR TITLE
Uses short names for buitlin types for diagnostics and bindings

### PR DESCRIPTION
**Pull request by MaxGraey**
Instead internat `~lib/string/String` or `~lib/string/Array<i32>` this PR simplify such names to `string` and `Array<i32>`

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: tests/compiler/bindings/esm.debug.d.ts [hunk 1]
```diff
@@ -141,11 +141,11 @@ declare interface __Record8<TOmittable> {
...
   /** @type `f64` */
   m: number | TOmittable;
-  /** @type `~lib/string/String | null` */
+  /** @type `string | null` */
   n: string | null | TOmittable;
-  /** @type `~lib/typedarray/Uint8Array | null` */
+  /** @type `Uint8Array | null` */
   o: Uint8Array | null | TOmittable;
-  /** @type `~lib/array/Array<~lib/string/String> | null` */
+  /** @type `Array<~lib/string/String> | null` */
```
Comment (MaxGraey) [inline tests/compiler/bindings/esm.debug.d.ts:148]: @dcodeIO it looks like generic parameters somehow handles differently?

**Comment by dcodeIO** [comment]
One way to think about this could be that *global* elements have a second internal name, e.g. `~lib/string/String` can also be looked up internally with just `String`. So by convention, if an element is in global scope, the short name could be used. This however differs slightly from what the PR does (looks a `~lib/` even if an element is not *global*) - perhaps change?

File: src/compiler.ts [hunk 2]
```diff
@@ -3570,7 +3574,9 @@ export class Compiler extends DiagnosticEmitter {
       }
       this.error(
         DiagnosticCode.Type_0_is_not_assignable_to_type_1,
-        reportNode.range, fromType.toString(), toType.toString()
+        reportNode.range,
+        fromType.toString(),
+        toType.toString()
```
Comment (dcodeIO) [inline src/compiler.ts:3579]: So far I used the convention (for readability of the overall file) to not expand this too much, as it doesn't really add much. I somewhat preferred the old way, or are there good reasons to make this more verbose?

**Comment by MaxGraey** [comment]
>  So by convention, if an element is in global scope, the short name could be used. This however differs slightly from what the PR does (looks a ~lib/ even if an element is not global) - perhaps change?

Hmm, interesting. But how I can check is this element global easily?

File: tests/compiler/bindings/esm.debug.d.ts (hunk 1 above)
Comment (dcodeIO) [reply tests/compiler/bindings/esm.debug.d.ts:148]: This happens at

https://github.com/AssemblyScript/assemblyscript/blob/d5599823cc7af3b858967477a4c3520f5c7784b0/src/types.ts#L471-L475

where the `internalName` of a concrete class is constructed from the full names upon resolve around

https://github.com/AssemblyScript/assemblyscript/blob/d5599823cc7af3b858967477a4c3520f5c7784b0/src/resolver.ts#L2891-L2900

The requirement here is that internal names are guaranteed to be unique, which I guess would still be the case if the shorter names are used according to the convention that it applies to *global* elements (see below)?

**Comment by dcodeIO** [comment]
Hmm, perhaps a check that `elementsByName.get(theShortName)` exists, i.e. that the assumption that the short name can be used holds.

**Pull request closed without merging**